### PR TITLE
vm, syscall: implement `enable_get_sysvar_syscall` feature

### PR DESCRIPTION
**Pull request by mjain-jump**

### CODE REVIEW

File: src/flamenco/vm/syscall/fd_vm_syscall_runtime.c [hunk 1]
```diff
@@ -175,6 +176,79 @@ fd_vm_syscall_sol_get_last_restart_slot_sysvar( /**/            void *  _vm,
   return FD_VM_SUCCESS;
 }
 
+/* https://github.com/anza-xyz/agave/blob/d8e9da5b5d02f6ee951e597dbb8e6c6fd5e32ff7/programs/bpf_loader/src/syscalls/sysvar.rs#L167-L232 */
```
Comment (0x0ece) [inline src/flamenco/vm/syscall/fd_vm_syscall_runtime.c:179]: maybe replace commit id with `v2.1.0`?

**Review by 0x0ece — APPROVED** [review]

File: src/flamenco/vm/syscall/fd_vm_syscall_runtime.c (hunk 1 above)
Comment (mjain-jump) [reply src/flamenco/vm/syscall/fd_vm_syscall_runtime.c]: yep good point, done

**Review by 0x0ece — APPROVED** [review]

File: src/flamenco/vm/syscall/fd_vm_syscall.h [hunk 2]
```diff
@@ -441,6 +441,41 @@ FD_VM_SYSCALL_DECL( sol_get_fees_sysvar              );
 FD_VM_SYSCALL_DECL( sol_get_rent_sysvar              );
 FD_VM_SYSCALL_DECL( sol_get_last_restart_slot_sysvar );
 
+/* syscall(FIXME) "sol_get_sysvar"
```
Comment (ripatel-fd) [inline src/flamenco/vm/syscall/fd_vm_syscall.h:444]: Why the FIXME here?

File: src/flamenco/vm/syscall/fd_vm_syscall_runtime.c [hunk 3]
```diff
@@ -175,6 +176,79 @@ fd_vm_syscall_sol_get_last_restart_slot_sysvar( /**/            void *  _vm,
...
+
+  if( FD_UNLIKELY( offset_length>sysvar_buf_len ) ) {
+    *_ret = 1UL;
+    return FD_VM_SUCCESS;
+  }
+
+  if( FD_UNLIKELY( sz==0UL ) ) {
+    *_ret = 0UL;
+    return FD_VM_SUCCESS;
+  }
```
Comment (ripatel-fd) [inline src/flamenco/vm/syscall/fd_vm_syscall_runtime.c:245]: You can remove this, `fd_memcpy` explicitly handles `sz==0` 
(But love the attention to detail :) )

Comment (ripatel-fd) [reply src/flamenco/vm/syscall/fd_vm_syscall_runtime.c:245]: (Manik found an LLVM bug so this branch has to stay here)

**Review by topointon-jump — DISMISSED** [review]
This is awesome, fantastic work

File: src/flamenco/vm/syscall/fd_vm_syscall_runtime.c [hunk 4]
```diff
@@ -175,6 +176,79 @@ fd_vm_syscall_sol_get_last_restart_slot_sysvar( /**/            void *  _vm,
...
+  FD_VM_CU_UPDATE( vm, fd_ulong_sat_add( FD_VM_SYSVAR_BASE_COST, fd_ulong_max( sysvar_buf_cost, FD_VM_MEM_OP_BASE_COST ) ) );
+
+  /* https://github.com/anza-xyz/agave/blob/v2.1.0/programs/bpf_loader/src/syscalls/sysvar.rs#L199-L200 */
+  const fd_pubkey_t * sysvar_id = FD_VM_MEM_HADDR_LD( vm, sysvar_id_vaddr, FD_VM_ALIGN_RUST_PUBKEY, FD_PUBKEY_FOOTPRINT );
+
+  /* https://github.com/anza-xyz/agave/blob/v2.1.0/programs/bpf_loader/src/syscalls/sysvar.rs#L202-L203 */
+  void * out_haddr = FD_VM_MEM_SLICE_HADDR_ST( vm, out_vaddr, FD_VM_ALIGN_RUST_U8, sz );
+
+  /* TODO: find the exact return error code
+     https://github.com/anza-xyz/agave/blob/v2.1.0/programs/bpf_loader/src/syscalls/sysvar.rs#L205-L208 */
```
[the remark this answers is not in the document]
Comment (mjain-jump) [reply src/flamenco/vm/syscall/fd_vm_syscall_runtime.c:202]: this is fixed, ill remove it and re-squash

**Review by topointon-jump — APPROVED** [review]

**Pull request merged**